spec/controllers/ops_controller/settings/common_spec.rb

```diff
@@ -239,6 +240,50 @@
           expect(assigns(:ntp_servers)).to eq("1.example.com, 2.example.com")
         end
       end
+
+      context 'get advanced config settings' do
+        it 'for selected server' do
+          _guid, miq_server, _zone = EvmSpecHelper.local_guid_miq_server_zone
+          enc_pass = MiqPassword.encrypt('pa$$word')
+          stub_settings(:http_proxy => {:default => {:host     => "proxy.example.com",
+                                                     :user     => "user",
+                                                     :password => enc_pass,
+                                                     :port     => 80}})
+          allow(controller).to receive(:x_node).and_return("svr-#{miq_server.id}")
+          controller.instance_variable_set(:@sb, :active_tab => 'settings_advanced')
+          controller.send(:settings_get_info, "svr-#{miq_server.id}")
+          expect(assigns(:edit)[:current]).to eq(:file_data => "---\n:http_proxy:\n  :default:\n    :host: proxy.example.com\n    :user: user\n    :password: #{enc_pass}\n    :port: 80\n")
+        end
+      end
+    end
+
+    describe '#settings_update_save' do
+      context "save config settings" do
+        it 'for selected server' do
+          _guid, miq_server, _zone = EvmSpecHelper.local_guid_miq_server_zone
+          allow(controller).to receive(:x_node).and_return("svr-#{miq_server.id}")
+          controller.instance_variable_set(:@sb,
+                                           :active_tab         => 'settings_advanced',
+                                           :selected_server_id => miq_server.id)
+          controller.instance_variable_set(:@_params,
+                                           :id => 'advanced')
+          data = {}
+          data.store_path(:api, :token_ttl, "1.day")
+          data = data.to_yaml
+          controller.instance_variable_set(:@edit,
+                                           :new     => {:file_data => data},
+                                           :current => {:file_data => data},
+                                           :key     => "settings_advanced_edit__#{miq_server.id}")
+          session[:edit] = assigns(:edit)
+          expect(controller).to receive(:render)
+          expect(Vmdb::Settings).to receive(:reload!).exactly(2).times
+
+          controller.send(:settings_update_save)
+          controller.send(:fetch_advanced_settings, miq_server)
+          expect(VMDB::Config.save_file(data, miq_server)).to be true
```

Comment:
The controller itself should be calling this, so why are you calling it in the test directly?